### PR TITLE
make authApi non throw exception when server returns 409

### LINKED ISSUE #131: Don't emit an error when an API that creates an account returns 409
## Kind (Required)

- Improvement

 ## Overview (Required)

- `POST /accounts` returns 409 if the user already exist on the server-side, so it's okay to suppress it and pass through the process.

 ## Links	

- https://github.com/DroidKaigi/conference-app-2021/blob/03d10c3b3f8414e18a7173ad4c15b8b4a057d2c7/data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt#L45

### PR DESCRIPTION
**Pull request by RyuNen344**
## Issue
- close #131

## Overview (Required)
- add handling to AuthApi
- I wanted throw appError in getOrElse statements, but appError extends RuntimeException.

## Links
- 

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt [hunk 1]
```diff
@@ -43,12 +49,16 @@ class AuthApi(
...
 
     private suspend fun registerToServer(createdIdToken: String) {
-        httpClient.post<String>("https://ssot-api-staging.an.r.appspot.com/accounts") {
-            headers {
-                set("Authorization", "Bearer $createdIdToken")
+        runCatching {
+            httpClient.post<String>("https://ssot-api-staging.an.r.appspot.com/accounts") {
+                header(HttpHeaders.Authorization, "Bearer $createdIdToken")
+                contentType(ContentType.Application.Json)
+                body = "{}"
```
Comment (RyuNen344) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt:56]: maybe , non need empty body...?

**Comment by takahirom** [comment]
👀

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt (hunk 1 above)
Comment (takahirom) [reply data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt:53]: When I tried it before, the server returned an error if I didn't send this "{}". If it doesn't seem to work, you can turn it off. 💭 
For the time being, this PR does not have to deal with it.  👍

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt [hunk 2]
```diff
@@ -43,12 +46,16 @@ class AuthApi(
     }
 
     private suspend fun registerToServer(createdIdToken: String) {
-        httpClient.post<String>("https://ssot-api-staging.an.r.appspot.com/accounts") {
-            headers {
-                set("Authorization", "Bearer $createdIdToken")
+        runCatching {
```
Comment (takahirom) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt:49]: 🆒

**Review by takahirom — APPROVED** [review]
LGTM 👍

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt (hunk 1 above)
Comment (RyuNen344) [reply data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt:53]: hmm.... OK!

**Pull request merged**